pyatv/raop/rtsp.py

```diff
@@ -153,12 +163,30 @@ async def announce(self) -> HttpResponse:
             channels=self.context.channels,
             sample_rate=self.context.sample_rate,
         )
-        return await self.exchange(
+
+        requires_password: bool = (password is not None)
+
+        response = await self.exchange(
             "ANNOUNCE",
             content_type="application/sdp",
             body=body,
+            allow_error=requires_password
         )
 
+        # Save the necessary data for password authentication
+        www_authenticate = response.headers.get("www-authenticate", None)
+        if response.code == 401 and www_authenticate and requires_password:
+            _, realm, _, nonce, _ = www_authenticate.split("\"")
+            self.context.digest_info = DigestInfo("pyatv", realm, password, nonce)
+
+            response = await self.exchange(
+                "ANNOUNCE",
+                content_type="application/sdp",
+                body=body,
+            )
+
+        return response
```

Comment:
If I understand the implementation correctly, we will just fall-though here and rely on the next call raising an exception in case a password is required but none is provided? Maybe we can be helpful and raise a proper exception in that case (with a string mentioning a password is required)?

Reply:
I'm not sure I'm understanding you correctly. If the user does not provide a password and a password is required
`allow_error` will be `false`. That means the first `ANNOUNCE` will raise an exception for the `AuthenticationError`.

Reply:
Ah, yes, I missed that! You are correct, that is handled good enough (I.e. like before), so I'm satisfied with that 👍